contrib/completion/bash/docker

```diff
@@ -3098,13 +3106,27 @@ _docker_service_update_and_create() {
 				__docker_complete_image_repos_and_tags
 				return
 				;;
+			--placement-pref-add|--placement-pref-rm)
+				COMPREPLY=( $( compgen -W "spread" -S = -- "$cur" ) )
+				__docker_nospace
+				return
+				;;
 			--secret-add|--secret-rm)
 				__docker_complete_secrets
 				return
 				;;
 		esac
 	fi
 
+	local strategy=$(__docker_map_key_of_current_option '--placement-pref|--placement-pref-add|--placement-pref-rm')
+	case "$strategy" in
+		spread)
+			COMPREPLY=( $( compgen -W "node.labels" -S . -- "${cur##*=}" ) )
```

Comment:
I don't know exactly what this is doing, but it's interesting to me that just `node.labels` is mentioned here. It would also be valid to spread on `engine.labels.<label>`.